Plugin.pm

```diff
@@ -1800,4 +1837,87 @@ sub _localDate {  # convert input date string in format YYYY-MM-DD to localized
 	return strftime(preferences('server')->get('shortdateFormat'), 0, 0, 0, $dt[2], $dt[1] - 1, $dt[0] - 1900);
 }
 
+sub addRecentSearch {
+	my $search = shift;
+	main::DEBUGLOG && $log->is_debug && $log->debug("++addRecentSearch");
+
+	my $list = $prefs->get('qobuz_recent_search') || [];
+
+	$list = [ grep { $_ ne $search } @$list ];
+
+	push @$list, $search;
+
+	$list = [ @$list[(-1 * MAX_RECENT)..-1] ] if scalar @$list > MAX_RECENT;
+
+	$prefs->set( 'qobuz_recent_search', $list );
+	main::DEBUGLOG && $log->is_debug && $log->debug("--addRecentSearch");
+	return;
+}
+
+sub _recentSearchesCLI {
+	my $request = shift;
+	my $client = $request->client;
+	main::DEBUGLOG && $log->is_debug && $log->debug("++_recentSearchesCLI");
+
+	# check this is the correct command.
+	if ($request->isNotCommand([['qobuz'], ['recentsearches']])) {
+		$request->setStatusBadDispatch();
+		return;
+	}
+
+	my $list = $prefs->get('qobuz_recent_search') || [];
+	my $del = $request->getParam('deleteMenu') || $request->getParam('delete') || 0;
+
+	if (!scalar @$list || $del >= scalar @$list) {
+		$log->error('Search item to delete is outside the history list!');
+		$request->setStatusBadParams();
+		return;
+	}
+
+	my $items = [];
+
+	if (defined $request->getParam('deleteMenu')) {
+		push @$items,
+		  {
+			text => cstring($client, 'DELETE') . cstring($client, 'COLON') . ' "' . ($list->[$del] || '') . '"',
+			actions => {
+				go => {
+					player => 0,
+					cmd    => ['qobuz', 'recentsearches' ],
+					params => {
+						delete => $del
+					},
+				},
+			},
+			nextWindow => 'parent',
+		  },
+		  {
+			text => 'Clear search history',
```

Comment:
Please use a string token here.

Reply:
Where are the system-wide translations? (I wonder if there are existing ones I can use)

Reply:
System wide are in LMS code in the root's `strings.txt`. If you can't find anything suitable there you can add your own.

Reply:
Nothing suitable in there, so I should add to the plugin's  strings.txt?

Reply:
Sure.